src/org/integr8ly/GitHubUtils.groovy

```diff
@@ -17,22 +17,20 @@ import org.kohsuke.github.PagedIterable
  * @param endpoint - Github endpoint to send the request to (https://developer.github.com/v3/apps/available-endpoints/)
  * @param httpMethod - HTTP method to be used for the request
  * @param requestBody - HTTP request body
- * @param ghApiTokenID - The credentials id where the github api token is stored in Jenkins
+ * @param credentialsID - The credentials id where the github user/password is stored in Jenkins
  * @param customHeaders - Additional headers to be used for the request
  * @param validResponseCodes - Valid HTTP response codes
  * @returns HTTP response
  */
-def ghApiRequest(String endpoint, String httpMethod = 'GET', String requestBody = null, String ghApiTokenID, customHeaders = [], validResponseCodes = '100:399') {
+def ghApiRequest(String endpoint, String httpMethod = 'GET', String requestBody = null, String credentialsID, validResponseCodes = '100:399') {
     def url = "https://api.github.com/${endpoint}"
-    withCredentials([string(credentialsId: ghApiTokenID, variable: 'gitToken')]) {
-        customHeaders.plus(["name": "Authorization", "value": "token ${env.gitToken}"])
-    }
 
     def response = httpRequest httpMode: httpMethod,
+                                authentication: credentialsID,
                                 contentType: 'APPLICATION_JSON',
                                 consoleLogResponseBody: true, 
                                 requestBody: requestBody,
-                                customHeaders: customHeaders,
+                                customHeaders: [],
```

Comment:
do we need this line at all?

Reply:
Not if we're not allowing it be set as one of the parameters.

Reply:
I've put back the customHeaders param now, should just default to an empty array